cloudevents/extensions/verifiability.md

```diff
@@ -0,0 +1,290 @@
+and `verificationmaterialtype` attributes. Consumers can choose to verify if
+the material.
+
+## Assumptions
+
+This proposal contains a few assumptions that will be highlighted here.
+
+1. SDKs will verify as early as possible which depends on the
+   verification implementation.
+2. Users manage their secrets, e.g. public key infrastructure (PKI).
+
+## Design
+
+Verifiability in CloudEvents consists of two steps:
+
+1. The producer of an event adds verification material to the message
+2. The consumer of an event MAY use the verification material to verify the
+   authenticity and integrity of the event
+
+
+The specifics of what the verification material looks like and how the
+verification is performed depend on the *verifiability implementation*.
+
+For example, in a public/private key based implementation the producer of an
+event would add a signature based on their private key as the verification
+material, and the consumer of an event could use the corresponding public key
+to check the if the signature matches the event.
+
+The flow looks like this:
+
+![Flow](./verifiability-flow.jpg)
+
+*Option 1*: useful for verifiability implementations outside of the CE SDK. The
+CE SDK merely passes on messages and is not involved in producing verification
+material or performing verifications and has no knowledge about the secrets
+that the verification is based on (e.g. private keys). Appropriate for tools in
+a closed ecosystem like an enterprise with special requirements that are not
+suitable for the CE SDK.
+
+*Option 2*: useful for verifiability implementations that are directly
+supported by the CE SDK. The CE SDK has to be provided with the secrets (e.g.
+private key for producer and public key for consumers) and will create the
+verification material on the producer side and also perform the verification on
+the consumer side. This is the appropriate choice for any type of tool that is
+used by other entities, for example open source.
+
+**Note:** If an intermediary party modifies an event, they are considered the
+producer of a new event. They MUST create an updated verification material and
+the consumer(s) MUST recognize them as a trustworthy producer.
+
+## Verifiability Implementations
+
+While this proposal is agnostic to the specific implementations, we set the
+following rules for verifiability implementations:
+
+A verifiability implementation **MUST**:
+
+* transport the verification material in the same message as the event
+* use the two relevant context attributes in this proposal
+
+A verifiability implementation **SHOULD**:
+
+* avoid depending on canonicalization (and its
+[unnecessarily large attack surface](https://github.com/secure-systems-lab/dsse/blob/master/background.md#motivation))
+* transport the verification material detached from the event payload
+(this helps avoid canonicalization)
+
+A verifiability implementation **MUST FAIL** on:
```

Comment:
I'm move all of the requirements (in this section) into the core part of the spec (into the "Attributes" section), so everything critical is in one section. Then these other sections are just helper sections to better understand why certain decisions were made and guidance on how to implement it.